src/wallet/rpcdump.cpp

```diff
@@ -398,7 +395,10 @@ UniValue importpubkey(const JSONRPCRequest& request)
 
     if (fRescan)
     {
-        pwalletMain->ScanForWalletTransactions(chainActive.Genesis(), true);
+        if (fPruneMode)
+          pwalletMain->ScanForWalletUTXOs();
+        else
+          pwalletMain->ScanForWalletTransactions(chainActive.Genesis(), true);
```

Comment:
I could definitely add it as another option to the RPC. I guess the benefit to the RPC caller would be that if they're importing a key which they already know is unspent, calling `ScanForWalletUTXOs` would be faster than calling `ScanForWalletTransactions`?